src/main/java/org/myrobotlab/service/TimeEncoder.java

```diff
@@ -31,28 +51,39 @@ public TimeEncoder(String n) {
   static public ServiceType getMetaData() {
 
     ServiceType meta = new ServiceType(TimeEncoder.class);
-    meta.addDescription("used as a general template");
+    meta.addDescription("general purpose timing encoder used in place of real hardware");
     meta.setAvailable(true);
     meta.setAvailable(false);
     meta.addCategory("general");
     return meta;
   }
 
   @Override // from EncoderController
-  public void attach(EncoderControl control, Integer pin) throws Exception {
+  public EncoderData publishEncoderData(EncoderData data) {
     // TODO Auto-generated method stub
-    
+    return null;

```

Comment:
this should return the data. not null.

Reply:
barely begun - I've just created the class ;)